Lib/idlelib/textview.py

```diff
@@ -28,52 +29,104 @@ def place(self, **kwargs):
-        self['relief'] = 'sunken'
-        self['height'] = 700
+        super().__init__(master, **kwargs)
 
-        self.text = text = Text(self, wrap=wrap, highlightthickness=0)
-        color_config(text)
-        text.grid(row=0, column=0, sticky=N+S+E+W)
+        @add_config_callback(self._config_callback)
+        class TextWithConfigHook(Text):
+            pass
+        self.text = TextWithConfigHook(self)
+        self.text.grid(row=0, column=0, sticky=NSEW)
         self.grid_rowconfigure(0, weight=1)
         self.grid_columnconfigure(0, weight=1)
-        text.insert(0.0, rawtext)
-        text['state'] = 'disabled'
-        text.focus_set()
 
         # vertical scrollbar
-        self.yscroll = yscroll = AutoHiddenScrollbar(self, orient=VERTICAL,
-                                                     takefocus=False,
-                                                     command=text.yview)
-        text['yscrollcommand'] = yscroll.set
-        yscroll.grid(row=0, column=1, sticky=N+S)
-
-        if wrap == 'none':
-            # horizontal scrollbar
-            self.xscroll = xscroll = AutoHiddenScrollbar(self, orient=HORIZONTAL,
-                                                         takefocus=False,
-                                                         command=text.xview)
-            text['xscrollcommand'] = xscroll.set
-            xscroll.grid(row=1, column=0, sticky=E+W)
+        self.yscroll = AutoHiddenScrollbar(self, orient=VERTICAL,
+                                           takefocus=False,
+                                           command=self.text.yview)
+        self.yscroll.grid(row=0, column=1, sticky=NS)
+        self.text['yscrollcommand'] = self.yscroll.set
+
+        # horizontal scrollbar
+        self.xscroll = None
+        self._set_wrapping(self.text.cget('wrap'))
+
+    def _config_callback(self, cnf):
+        if 'wrap' in cnf:
+            self._set_wrapping(cnf['wrap'])
+
+    def _set_wrapping(self, wrap):
+        """show/hide the horizontal scrollbar as per the 'wrap' setting"""
+        # show the scrollbar only when wrap is set to NONE
+        if wrap == NONE and self.xscroll is None:
+            self.xscroll = AutoHiddenScrollbar(self, orient=HORIZONTAL,
+                                               takefocus=False,
+                                               command=self.text.xview)
+            self.xscroll.grid(row=1, column=0, sticky=EW)
+            self.text['xscrollcommand'] = self.xscroll.set
+        elif wrap != NONE and self.xscroll is not None:
+            self.text['xscrollcommand'] = ''
+            self.xscroll.grid_forget()
+            self.xscroll.destroy()
+            self.xscroll = None
 
 
 class ViewFrame(Frame):
     "Display TextFrame and Close button."
     def __init__(self, parent, text, wrap='word'):
+        """Create a frame for viewing text with a "Close" button.
+
+        parent - parent widget for this frame
+        text - text to display
```

Comment:
It was previously "rawtext", which I dislike.

FWIW I find the variable `text` for a text widget, rather than an actual string of text, confusing. I realize that ship has sailed, though.

I suggest 'contents' or 'text_string'.